### ISSUE TITLE
Question about the randomness of sample_points_from_meshes function

### ISSUE THREAD

**Issue by cuge1995** [facebookresearch/pytorch3d#597]
I sample point from a mesh as follows:
```python
In [3]: src_mesh = ico_sphere(4, 'cuda')

In [4]: c = sample_points_from_meshes(src_mesh, 1024)

In [5]: c
Out[5]:
tensor([[[ 0.4231, -0.8899, -0.1653],
         [ 0.2461,  0.2795,  0.9272],
         [ 0.4509, -0.0734, -0.8885],
         ...,
         [-0.3945,  0.5399, -0.7428],
         [ 0.0130,  0.4026, -0.9148],
         [-0.6370, -0.7540, -0.1554]]], device='cuda:0')

In [6]:

In [6]: d = sample_points_from_meshes(src_mesh, 1024)

In [7]: d
Out[7]:
tensor([[[-0.0905,  0.7744,  0.6255],
         [-0.0908,  0.6709, -0.7348],
         [-0.9539,  0.2925,  0.0481],
         ...,
         [ 0.2986,  0.3999, -0.8657],
         [-0.6530, -0.1851, -0.7336],
         [-0.5106,  0.7976, -0.3185]]], device='cuda:0')

In [8]: e = sample_points_from_meshes(src_mesh, 1024)

In [9]: e
Out[9]:
tensor([[[-0.3843, -0.9021, -0.1936],
         [ 0.3511, -0.4152,  0.8382],
         [ 0.3759, -0.8644, -0.3316],
         ...,
         [ 0.5293, -0.8232, -0.2013],
         [ 0.6844, -0.3636,  0.6310],
         [-0.0315, -0.3450,  0.9372]]], device='cuda:0')
```
It looks like every sampling is different, so, how this random came from?

**Comment by bottler** [comment]
PyTorch itself contains random number generation facilities, with functions like `multinomial`, `rand`, and `randn`. The randomness in `sample_points_from_meshes` is coming from there. If you need to get the same result every time, look at `torch.manual_seed`, and more generally look at the [PyTorch reproducibility page](https://pytorch.org/docs/stable/notes/randomness.html).

**Comment by cuge1995** [comment]
> PyTorch itself contains random number generation facilities, with functions like `multinomial`, `rand`, and `randn`. The randomness in `sample_points_from_meshes` is coming from there. If you need to get the same result every time, look at `torch.manual_seed`, and more generally look at the [PyTorch reproducibility page](https://pytorch.org/docs/stable/notes/randomness.html).

I don't think so.
```python
In [1]: import torch

In [2]: torch.manual_seed(1234)
Out[2]: <torch._C.Generator at 0x7f5177adfd20>

In [3]: from pytorch3d.ops import sample_points_from_meshes
   ...: from pytorch3d.structures import Meshes
   ...: from pytorch3d.structures import join_meshes_as_batch
   ...: from pytorch3d.utils import ico_sphere

In [4]: src_mesh = ico_sphere(4, 'cuda')

In [5]: c = sample_points_from_meshes(src_mesh, 1024)

In [6]: c
Out[6]:
tensor([[[ 0.6870, -0.7173,  0.1087],
         [-0.7464, -0.0515,  0.6629],
         [ 0.2249, -0.8763,  0.4239],
         ...,
         [ 0.3335,  0.7749,  0.5357],
         [-0.7447,  0.2259, -0.6267],
         [ 0.8396, -0.0225,  0.5414]]], device='cuda:0')

In [7]: d = sample_points_from_meshes(src_mesh, 1024)

In [8]: d
Out[8]:
tensor([[[-0.6031, -0.5719, -0.5542],
         [-0.2445,  0.4148,  0.8755],
         [-0.8281, -0.1151, -0.5481],
         ...,
         [ 0.5711, -0.3554,  0.7389],
         [ 0.2558,  0.7163,  0.6482],
         [-0.9860,  0.1504, -0.0585]]], device='cuda:0')

In [9]: e = sample_points_from_meshes(src_mesh, 1024)

In [10]: e
Out[10]:
tensor([[[-0.3703, -0.6170, -0.6939],
         [ 0.8227,  0.4982,  0.2703],
         [ 0.6486, -0.7599, -0.0139],
         ...,
         [ 0.9270, -0.1623,  0.3356],
         [ 0.1038,  0.5621,  0.8200],
         [ 0.9757, -0.2062,  0.0629]]], device='cuda:0')
```

**Comment by bottler** [comment]
Try setting the seed with `torch.manual_seed(1234)` before every call to `sample_points_from_meshes`, not just the first one.

**Comment by bottler** [comment]
Is everything clear now?

**Comment by gkioxari** [comment]
No response. Closing this issue. Feel free to reopen.
